e2e/frontend/configEditor.spec.ts

```diff
@@ -0,0 +1,55 @@
+import { expect, test } from '@grafana/plugin-e2e';
+
+const PLUGIN_UID = 'grafana-surrealdb-datasource';
+
+test.describe('Config Editor', () => {
+  test('invalid credentials should return an error', async ({ createDataSourceConfigPage, page }) => {
+    const configPage = await createDataSourceConfigPage({ type: PLUGIN_UID });
+    await page.getByPlaceholder('ws://localhost:8000/rpc').fill('ws://localhost:8000/rpc');
+    await expect(configPage.saveAndTest()).not.toBeOK();
+  });
+
+  test('valid credentials should return a 200 status code', async ({ createDataSourceConfigPage, page }) => {
+    const configPage = await createDataSourceConfigPage({ type: PLUGIN_UID });
+    configPage.mockHealthCheckResponse({ status: 200 });
+
+    await page.getByPlaceholder('ws://localhost:8000/rpc').fill('ws://localhost:8000/rpc');
+    await page.getByPlaceholder('Database name').fill('test');
+    await page.getByPlaceholder('Namespace').fill('test');
+    await page.getByPlaceholder('Username').fill('root');
+    await page.getByPlaceholder('Password').fill('test');
+
+    await expect(configPage.saveAndTest()).toBeOK();
+  });
+
+  test('valid credentials should display a success alert on the page', async ({ createDataSourceConfigPage, page }) => {
```

Comment:
Can this test be combined with the one above?

Reply:
IMO these are separate tests and should remain as such.